src/components/ha-icon.ts

```diff
@@ -109,25 +113,34 @@ export class HaIcon extends LitElement {
     if (!this.icon) {
       return html``;
     }
-    if (this._noMdi) {
+    if (this._legacy) {
       return html`<iron-icon .icon=${this.icon}></iron-icon>`;
     }
-    return html`<ha-svg-icon .path=${this._path}></ha-svg-icon>`;
+    return html`<ha-svg-icon
+      .path=${this._path}
+      .viewBox=${this._viewBox}
```

Comment:
By setting the default value in the render mode for viewBox, you can remove the default from both `updated` and `setCustomPath`
 
```suggestion
      .viewBox=${this._viewBox || "0 0 24 24"}
```